v2/reader.go

```diff
@@ -121,3 +124,62 @@ func ReadVersion(r io.Reader) (version uint64, err error) {
 	}
 	return header.Version, nil
 }
+
+var _ io.ReaderAt = (*readSeekerAt)(nil)
+
+type readSeekerAt struct {
+	rs io.ReadSeeker
+	mu sync.RWMutex
```

Comment:
Shouldn't this be a Mutex instead of an RWMutex ?

Reply:
doh indeed 🍻